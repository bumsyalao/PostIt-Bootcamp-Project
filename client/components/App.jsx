 /*./client/components/App.js */

import React from 'react';

class App extends React.Component{
  render() {
    return (
      <div style ={{textAlign: 'center'}}>
        <h1>Welcome to Post-It</h1>
      </div>);
  }
}