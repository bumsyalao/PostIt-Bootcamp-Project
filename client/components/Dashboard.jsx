import React from 'react';
import { Link } from 'react-router-dom';

const Dashboard = () => (
        <div className="section no-pad-bot" id="index-banner">
          <div className="container">
            <br></br>
            <div className="center">
            <img src ="https://www.embracingspirituality.com/wp-content/uploads/wpforo/default_attachments/1489579326-welcome.jpg"/>
            </div>
            <div className="row center">
              <h3 className="header col s12 light">Share Anything, With Anyone!</h3>
            </div>
            <div className="row center">
              <a href="register1.html" id="download-button" className="btn-large waves-effect waves-light red lighten-2">
                <Link to="/signup">Get Started</Link> </a>
            </div>
            <br></br>

          </div>
        </div>
    );

export default Dashboard;

