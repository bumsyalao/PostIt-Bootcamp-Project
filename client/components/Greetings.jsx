import React from 'react';

class Greetings extends React.Component {
   render() { 
    return (
        <div className="section no-pad-bot" id="index-banner">
          <div className="container">
            <br></br>
            <div className="center">
            <img src ="https://www.embracingspirituality.com/wp-content/uploads/wpforo/default_attachments/1489579326-welcome.jpg"/>
            </div>
            <div className="row center">
              <h3 className="header col s12 light">Share Anything, With Anyone!</h3>
            </div>
            <div className="row center">
              <a href="register1.html" id="download-button" className="btn-large waves-effect waves-light red lighten-2">Get Started</a>
            </div>
            <br></br>
          </div>
        </div>

    );
 }
}

export default Greetings;