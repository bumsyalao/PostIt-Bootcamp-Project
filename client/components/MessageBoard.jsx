// import React, { Component } from 'react';

// class MessageBoard extends Component {
//   render(){
//       if (localStorage.getItem("user") === null) {
//   // eslint-disable-next-line
//   this.props.history.push(`/siginin`);
//   return;
//   } else {
//     return (

//     )
//   }
// }