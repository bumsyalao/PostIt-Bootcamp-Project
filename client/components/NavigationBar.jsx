import React from 'react';
import { Link } from 'react-router-dom';

const NavigationBar = () => (
      <nav>
          <div className="nav-wrapper container">
              <Link to="/">
              <a id="logo-container" className="brand-logo">POST-IT</a>
              </Link>
            <ul className="right">
              <li><a href="login.html">Login</a></li>
            </ul>
            <ul className="right">
              <li><Link to="/signup">Register</Link></li>
            </ul>
            <ul id="nav-mobile" className="side-nav">
              <li><a href="#">Navbar Link</a></li>
            </ul>
            <a href="#" data-activates="nav-mobile" className="button-collapse"><i className="mdi-navigation-menu"></i></a>
          </div>
        </nav>
    );

export default NavigationBar;