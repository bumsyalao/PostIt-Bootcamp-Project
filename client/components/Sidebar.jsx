import React from 'react';
import { Link } from 'react-router-dom';
import feminist from '../images/feminist.png';


const SideBar = () => (
  <div>
    <ul id="slide-out" className="side-nav fixed #e57373 red lighten-2">
      <li>
        <div className="user-view">
          <div className="side-logo center">
            <img src={feminist} width="50%" height="50%" />
          </div>
        </div>
      </li>
      <ul className="collapsible " data-collapsible="accordion">
        <li>
          <div className="collapsible-header" id="document-header">
            <i className="material-icons">group</i>Groups
          </div>
          <div className="collapsible-body #ffebee red lighten-5" id="groups">
            <ul>
              <li>
                <Link to="/homepage/create-group" id="create-group">
                  <i className="material-icons">group_add</i>
                  Create Group
                </Link>
              </li>
              <li>
                <Link to="/homepage/groups">
                  <i className="material-icons">groups</i>
                  All Groups
                </Link>
              </li>
            </ul>
          </div>
        </li>
      </ul>
      <ul className="collapsible " data-collapsible="accordion">
        <li>
          <div className="collapsible-header" id="document-header">
            <i className="material-icons">person</i>Users
          </div>
          <div className="collapsible-body #ffebee red lighten-5" id="groups">
            <ul>
              <li>
                <Link to="/homepage/user-profile" id="create-group">
                  <i className="material-icons">face</i>
                  Profile
                </Link>
              </li>
              <li>
                <Link to="/homepage/search-users">
                  <i className="material-icons">search</i>
                  Search User
                </Link>
              </li>
            </ul>
          </div>
        </li>
      </ul>
    </ul>
  </div>
);

export default SideBar;
