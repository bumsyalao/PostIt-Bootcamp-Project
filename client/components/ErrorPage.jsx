import React from 'react';
import NavigationBar from './NavigationBar';


const ErrorPage = () => (
  <div>
    <NavigationBar />
    <h4 className="error-page"> Error 404: Page Not Found </h4>
  </div>

);

export default ErrorPage;
