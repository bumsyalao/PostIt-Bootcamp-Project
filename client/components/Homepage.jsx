import React from 'react';
import { Route, Switch, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import NavigationBar from './NavigationBar';
import Sidebar from './Sidebar';
import CreateGroup from './Group/CreateGroup';
import ListGroup from './Group/ListGroup';
import GroupChat from './Group/GroupChat';
import UsersPage from '../components/Users/UsersPage';
import UserProfile from '../components/Users/UserProfile';
import Welcomepage from './Welcomepage';


/**
 * @class Homepage
 * @extends {React.Component}
 */
class Homepage extends React.Component {

  /**
   *
   * Check if user is authenticated
   * Toast access message
   * Intialize materialize collapsible
   *
   * @memberOf Homepage
   */
  componentDidMount() {
    if (!this.props.access.isAuthenticated) {
      this.props.history.push('/signin');
      Materialize.toast('Please SignIn or Register', 5000, 'red');
    }
    $('.collapsible').collapsible(); // eslint-disable-line
  }

  /**
   * Renders Homepage
   * @returns Homepage
   * @memberOf Homepage
   */
  render() {
    return (
      <div>
        <Sidebar />
        <div className="homepage">
          <NavigationBar match={this.props.match} />
          <Switch>
            <Route
              path={`${this.props.match.url}/welcome-page`}
              component={Welcomepage}
            />
            <Route
              path={`${this.props.match.url}/create-group`}
              component={CreateGroup}
            />
            <Route
              path={`${this.props.match.url}/groups`}
              component={ListGroup}
            />
            <Route
              path={`${this.props.match.url}/view-group/:groupId`}
              component={GroupChat}
            />
            <Route
              path={`${this.props.match.url}/search-users`}
              component={UsersPage}
            />
            <Route
              path={`${this.props.match.url}/user-profile`}
              component={UserProfile}
            />
          </Switch>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => (
  {
    access: state.access
  }
);

export default connect(mapStateToProps, null)(withRouter(Homepage));

