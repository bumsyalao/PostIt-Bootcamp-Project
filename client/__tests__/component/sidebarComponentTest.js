/* global expect jest test */
import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import adapter from 'enzyme-adapter-react-16';
// import feminist from '../../images/feminist.png';
import { Sidebar } from '../../components/Sidebar';

configure({ adapter: new adapter() });

describe('Sidebar component', () => {
  const spy = jest.fn();
  beforeEach(() => {
    global.Materialize = { toast: jest.fn() };
  });
  afterEach(() => {
    spy.mockReset();
  });

  const component = shallow(
    <Sidebar />

  );

  test('should match the Sidebar component snapshot', () => {
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
