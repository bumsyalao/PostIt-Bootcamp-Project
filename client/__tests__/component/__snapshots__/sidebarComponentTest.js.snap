// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component should match the Sidebar component snapshot 1`] = `
<div>
  <ul
    className="side-nav fixed #e57373 red lighten-2"
    id="slide-out"
  >
    <li>
      <div
        className="user-view"
      >
        <div
          className="side-logo center"
        >
          <img
            height="80%"
            src={
              Object {
                "process": [Function],
              }
            }
            width="80%"
          />
          <h5>
             POST-IT 
          </h5>
        </div>
      </div>
    </li>
    <ul
      className="collapsible "
      data-collapsible="accordion"
    >
      <li>
        <div
          className="collapsible-header"
          id="document-header"
        >
          <i
            className="material-icons"
          >
            group
          </i>
          Groups
        </div>
        <div
          className="collapsible-body #ffebee red lighten-5"
          id="groups"
        >
          <ul>
            <li>
              <Link
                id="create-group"
                replace={false}
                to="/homepage/create-group"
              >
                <i
                  className="material-icons"
                >
                  group_add
                </i>
                Create Group
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/homepage/groups"
              >
                <i
                  className="material-icons"
                >
                  groups
                </i>
                All Groups
              </Link>
            </li>
          </ul>
        </div>
      </li>
    </ul>
    <ul
      className="collapsible "
      data-collapsible="accordion"
    >
      <li>
        <div
          className="collapsible-header"
          id="document-header"
        >
          <i
            className="material-icons"
          >
            person
          </i>
          Users
        </div>
        <div
          className="collapsible-body #ffebee red lighten-5"
          id="groups"
        >
          <ul>
            <li>
              <Link
                id="create-group"
                replace={false}
                to="/homepage/user-profile"
              >
                <i
                  className="material-icons"
                >
                  face
                </i>
                Profile
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/homepage/search-users"
              >
                <i
                  className="material-icons"
                >
                  search
                </i>
                Search User
              </Link>
            </li>
          </ul>
        </div>
      </li>
    </ul>
    <ul
      className="collapsible "
      data-collapsible="accordion"
    >
      <li>
        <div
          className="collapsible-header"
          id="document-header"
        >
          <i
            className="material-icons"
          >
            power_settings_new
          </i>
          <a
            href="#"
            onClick={[Function]}
          >
            Log Out 
          </a>
        </div>
      </li>
    </ul>
  </ul>
</div>
`;
